### ISSUE TITLE
Incorrect match results

### ISSUE THREAD

**Issue by joachimschmidt557** [tiehuis/zig-regex#5]
Example to reproduce:

```zig
const std = @import("std");

const regex = @import("zig-regex/src/regex.zig");

pub fn main() !void {
    var gpa = std.heap.GeneralPurposeAllocator(.{}){};
    const allocator = &gpa.allocator;
    var re = try regex.Regex.compile(allocator, ".*emacs.*");

    std.debug.print("{}\n", .{try re.match("emacs-packages.nix")});
}
```

Output:

```
false
```

I'll try to look into what is causing this issue.

**Comment by kivikakk** [comment]
It appears to be because it invokes `VmBacktrack` instead of `VmPike` — I think it's not backtracking correctly. If I force it to use `VmPike` it matches correctly.

**Comment by kivikakk** [comment]
Concretely, here's how to get this returning `true`: https://github.com/kivikakk/zig-regex/commit/c81b4e548d0b24ca39783ba6e4c9f3554518361c

**Comment by kivikakk** [comment]
This is fixed by #9.
